notification/notification.go

```diff
@@ -0,0 +1,161 @@
+func NewCommentUpdated(commentID string) Message {
+	return Message{MessageID: uuid.NewV4(), MessageType: "comment.update", TargetID: commentID}
+}
+
+func setCurrentIdentity(ctx context.Context, msg *Message) {
+	currentUserIdentityID, err := login.ContextIdentity(ctx)
+	if err != nil {
+		uID := currentUserIdentityID.String()
+		msg.UserID = &uID
+	}
+}
+
+// DevNullChannel is the default configured channel. It does nothing.
+type DevNullChannel struct{}
+
+// Send NO-OP
+func (d *DevNullChannel) Send(context.Context, Message) {}
+
+// ServiceConfiguration holds configuration options required to interact with the fabric8-notification API
+type ServiceConfiguration interface {
+	GetNotificationServiceURL() string
+}
+
+// Service is a simple client Channel to the fabric8-notification service
+type Service struct {
+	config ServiceConfiguration
+}
+
+func validateConfig(config ServiceConfiguration) error {
+	_, err := url.Parse(config.GetNotificationServiceURL())
+	if err != nil {
+		return fmt.Errorf("Invalid NotificationServiceURL %v cause %v", config.GetNotificationServiceURL(), err.Error())
+	}
+	return nil
+}
+
+// NewServiceChannel sends notification messages to the fabric8-notification service
+func NewServiceChannel(config ServiceConfiguration) (Channel, error) {
+	err := validateConfig(config)
+	if err != nil {
+		return nil, err
+	}
+	return &Service{config: config}, nil
+}
+
+// Send invokes the fabric8-notification API
+func (s *Service) Send(ctx context.Context, msg Message) {
+	go func(ctx context.Context, msg Message) {
+		setCurrentIdentity(ctx, &msg)
+
+		u, err := url.Parse(s.config.GetNotificationServiceURL())
+		if err != nil {
+			log.Error(ctx, map[string]interface{}{
+				"err": err,
+			}, "unable to parse GetNotificationServiceURL")
+		}
+
+		cl := client.New(goaclient.HTTPClientDoer(http.DefaultClient))
+		cl.Host = u.Host
+		cl.Scheme = u.Scheme
+		cl.SetJWTSigner(goasupport.NewForwardSigner(ctx))
+
+		msgID := goauuid.UUID(msg.MessageID)
+
+		resp, err := cl.SendNotify(
+			ctx,
+			client.SendNotifyPath(),
+			&client.SendNotifyPayload{
+				Data: &client.Notification{
+					Type: "notifications",
+					ID:   &msgID,
+					Attributes: &client.NotificationAttributes{
+						Type: msg.MessageType,
+						ID:   msg.TargetID,
+					},
+				},
+			},
+		)
+		if err != nil {
+			log.Error(ctx, map[string]interface{}{
+				"message_id": msg.MessageID,
+				"type":       msg.MessageType,
+				"target_id":  msg.TargetID,
+				"err":        err,
+			}, "unable to send notification")
+		} else if resp.StatusCode >= 400 {
+			log.Error(ctx, map[string]interface{}{
+				"status":     resp.StatusCode,
+				"message_id": msg.MessageID,
+				"type":       msg.MessageType,
+				"target_id":  msg.TargetID,
+				"err":        err,
+			}, "unexpected response code")
+		}
+		defer resp.Body.Close()
+
+	}(ctx, msg)
+}
+
+// TestNotificationChannel is a simple Sender impl that records the notifications for later verification
```

Comment:
moved to test package https://github.com/fabric8-services/fabric8-wit/pull/1557/commits/4cf11c4e178132a85613f1ed8a175391d0922124